AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Models/AWSCognitoUserPoolTokens.swift

```diff
@@ -38,6 +38,32 @@ public struct AWSCognitoUserPoolTokens: AuthCognitoTokens {
         self.expiration = expiration
     }
 
+    public init(idToken: String,
+                accessToken: String,
+                refreshToken: String,
+                expiresIn: Int? = nil) {
+
+        if let expiresIn =  expiresIn {
+            self.init(idToken: idToken,
+                      accessToken: accessToken,
+                      refreshToken: refreshToken,
+                      expiresIn: expiresIn)
+        } else {
+            // If for some reason, failed to extract "exp" value, use the value as zero and don't block the user
+            var expirationDoubleValue: Double = 0
+            if let idTokenClaims = try? AWSAuthService().getTokenClaims(tokenString: idToken).get(),
+               let accessTokenClaims = try? AWSAuthService().getTokenClaims(tokenString: accessToken).get(),
```

Comment:
what if you can get one "exp" from one token, idToken or accessToken, but not the other? Would that be such a case? If so, maybe we can split it out as 0 by default, idToken's exp if accessToken doesn't exist, accessToken's exp if idToken doesn't exist, and then the min of the two if both exists. Just a thought, not sure if this there is such a scenario of token claims data.

Reply:
Nice.. Ideally this won't happen and it would never reach here. But updated the logic..